pkg/controller/utils/elasticsearch.go

```diff
@@ -62,3 +103,271 @@ func GetElasticsearchClusterConfig(ctx context.Context, cli client.Client) (*ren
 
 	return render.NewElasticsearchClusterConfigFromConfigMap(configMap)
 }
+
+type ElasticClient interface {
+	SetILMPolicies(client.Client, context.Context, *operatorv1.LogStorage) error
+}
+
+type EsClient struct {
+	client *elastic.Client
+	lock   sync.Mutex
+}
+
+// SetILMPolicies creates ILM policies for each timeseries based index using the retention period and storage size in LogStorage
+func (es *EsClient) SetILMPolicies(client client.Client, ctx context.Context, ls *operatorv1.LogStorage) error {
+	if err := es.createElasticClient(client, ctx); err != nil {
+		return err
+	}
+	policyList := es.listILMPolicies(ls)
+	return es.createOrUpdatePolicies(ctx, policyList)
+}
+
+func (es *EsClient) createElasticClient(client client.Client, ctx context.Context) error {
+	if es.client == nil {
+		user, password, root, err := getClientCredentials(client, ctx)
+		if err != nil {
+			return err
+		}
+		h := &http.Client{
+			Transport: &http.Transport{TLSClientConfig: &tls.Config{RootCAs: root}},
+		}
+
+		es.lock.Lock()
+		defer es.lock.Unlock()
+		esClient, err := newElastic(user, password, render.ElasticsearchHTTPSEndpoint, h)
```

Comment:
I think there is no need to have a separate newElastic func. The contents could just live here, together with the code that creates the http client.

Reply:
That's needed for test to create a mock client. If you look at the elasticsearch_tests.go, it create a mock http client that intercepts call to elasticsearch and validates what is being sent.